openedx/core/lib/api/permissions.py

```diff
@@ -74,3 +74,11 @@ def has_permission(self, request, view):
             return True
 
         return super(IsUserInUrlOrStaff, self).has_permission(request, view)
+
+
+class IsStaffOrReadOnly(permissions.BasePermission):
+    """
+    Permission that checks to see if the user is staff, permitting only read-only access if they are not.
+    """
+    def has_permission(self, request, view):
+        return request.user.is_staff or request.method in permissions.SAFE_METHODS
```

Comment:
To clarify about why some APIs want to use is_staff only instead of course staff, there are APIs that are not tied to a particular course. This was the case for student account settings and permissions.